src/wallet/wallet.h

```diff
@@ -607,10 +651,14 @@ class CWalletTx
      * >0 : is a coinbase transaction which matures in this many blocks
      */
     int GetBlocksToMaturity(interfaces::Chain::Lock& locked_chain) const;
-    bool hashUnset() const { return (hashBlock.IsNull() || hashBlock == ABANDON_HASH); }
-    bool isAbandoned() const { return (hashBlock == ABANDON_HASH); }
-    void setAbandoned() { hashBlock = ABANDON_HASH; }
-
+    bool isAbandoned() const { return tx_state.state == CWalletTx::ABANDONED; }
+    void setAbandoned() { tx_state.state = CWalletTx::ABANDONED; }
+    bool isConflicted() const { return tx_state.state == CWalletTx::CONFLICTED; }
+    void setConflicted() { tx_state.state = CWalletTx::CONFLICTED; }
+    bool isUnconfirmed() const { return tx_state.state == CWalletTx::UNCONFIRMED; }
+    void setUnconfirmed() { tx_state.state = CWalletTx::UNCONFIRMED; }
+    bool isConfirmed() const { return tx_state.state == CWalletTx::CONFIRMED; }
```

Comment:
`isConfirmed()` does not seem to be tested as the others. Add a test? :-)

Reply:
Removed for now, I've added it for further changes were ABANDON state was dual-set with other states, but drop if for now, would have been too big changes at once.